tests/rados/test_bug_fixes.py

```diff
@@ -298,3 +302,137 @@ def run(ceph_cluster, **kw):
+            "\n\t 1. Deploy a cluster with no default values of 'bluestore_min_alloc_size' and "
+            "'bluefs_shared_alloc_size'"
+            "\n\t 2. Fetch the current active value of 'bluestore_min_alloc_size'"
+            "\n\t 3. Choose one OSD from each OSD node at random"
+            "\n\t 4. Set the value of 'bluefs_shared_alloc_size' to half of 'bluestore_min_alloc_size' "
+            "for each chosen OSD"
+            "\n\t 5. Restart Orch OSD service"
+            "\n\t 6. Fill the cluster till 70%"
+            "\n\t 7. Ensure restart of Orch OSD service is successful"
+        )
+
+        log.info(doc)
+        log.info(
+            "Running test to verify OSD resiliency when 'bluefs_shared_alloc_size'"
+            " is below 'bluestore_min_alloc_size'"
+        )
+
+        try:
+            _pool_name = "test-bluefs-shared"
+
+            # get the list of OSDs on the cluster
+            osd_list = rados_obj.run_ceph_command(cmd="ceph osd ls")
+            log.debug(f"List of active OSDs: \n{osd_list}")
+
+            # check the current value of 'bluestore_min_alloc_size'
+            min_alloc_size = int(
+                mon_obj.show_config(
+                    daemon="osd", id=osd_list[0], param="bluestore_min_alloc_size_hdd"
+                )
+            )
+            log.info(
+                f"Current persistent value of 'bluestore_min_alloc_size_hdd' is {min_alloc_size}"
+            )
+            min_alloc_size_ssd = int(
+                mon_obj.show_config(
+                    daemon="osd", id=osd_list[0], param="bluestore_min_alloc_size_ssd"
+                )
+            )
+            log.info(
+                f"Current persistent value of 'bluestore_min_alloc_size_ssd' is {min_alloc_size}"
+            )
+
+            # cherry-pick one OSD ID from each osd node
+            test_osd_list = [
+                rados_obj.collect_osd_daemon_ids(osd_node=node)[0] for node in osd_nodes
+            ]
+
+            log.info(
+                f"OSDs for which bluefs_shared_alloc_size will be modified: {test_osd_list}"
+            )
+            shared_alloc_size = int(min_alloc_size / 2)
+            log.info(
+                f"bluefs_shared_alloc_size parameter will be set to - {shared_alloc_size}"
+            )
+
+            for osd_id in test_osd_list:
+                mon_obj.set_config(
+                    section=f"osd.{osd_id}",
+                    name="bluefs_shared_alloc_size",
+                    value=shared_alloc_size,
+                )
+
+            time.sleep(5)
+
+            # restart OSD orch service
+            assert rados_obj.restart_daemon_services(daemon="osd")
+
+            # determine how much % cluster is already filled
+            current_fill_ratio = rados_obj.get_cephdf_stats()["stats"][
+                "total_used_raw_ratio"
+            ]
+
+            # determine the number of objects to be written to the pool
```

Comment:
We are using this workflow of filling given pool up to X%. 

Should be fine for now, but Please create a backlog task for creating a new method for the above function.

Reply:
Yes, we are calculating the current fill % for the whole cluster in line # 383
Here is the output to further explain the code

```
--- RAW STORAGE ---
CLASS     SIZE    AVAIL     USED  RAW USED  %RAW USED
hdd    475 GiB  473 GiB  2.2 GiB   2.2 GiB       0.47
TOTAL  475 GiB  473 GiB  2.2 GiB   2.2 GiB       0.47
 
--- POOLS ---
POOL                   ID  PGS   STORED  OBJECTS     USED  %USED  MAX AVAIL
device_health_metrics   1    1      0 B        0      0 B      0    157 GiB
cephfs.cephfs.meta      2   32  2.3 KiB       22   96 KiB      0    157 GiB
cephfs.cephfs.data      3   32      0 B        0      0 B      0    157 GiB
repool_1                7   32  4.3 KiB      100  1.2 MiB      0    157 GiB
repool_2                8   32  4.3 KiB      100  800 KiB      0    235 GiB
ercpool_1               9   32  3.1 MiB      100  4.3 MiB      0    342 GiB
ercpool_2              10   32  3.1 MiB      100  4.7 MiB      0    314 GiB
[ceph: root@ceph-hakumar-6l6i6b-node1-installer /]# ceph df -f json-pretty

{
    "stats": {
        "total_bytes": 509947674624,
        "total_avail_bytes": 507550515200,
        "total_used_bytes": 2397159424,
        "total_used_raw_bytes": 2397159424,
        "total_used_raw_ratio": 0.0047007948160171509,
        "num_osds": 19,
        "num_per_pool_osds": 19,
        "num_per_pool_omap_osds": 19
    }
```
 
Once we have the current cluster fill %, we subtract the value from 70% and determine how many 16MB objects we would need to write to the pool to achieve the desired % at pool level

Reply:
Got it. Thanks. Plean to make this a reusable method.